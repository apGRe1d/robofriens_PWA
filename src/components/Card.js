import React from 'react';

// class Card extends React.PureComponent {
//   constructor(props) {
//     super(props);
//   }
//
//   render() {
//     const {id, name, email} = this.props;
//     return (
//       <div className='tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5'>
//         <img alt='robots' src={`https://robohash.org/${id}?size=200x200`} />
//         <div>
//           <h2>{name}</h2>
//           <p>{email}</p>
//         </div>
//       </div>
//     );
//   }
// }

const Card = ({ name, email, id }) => {
  return (
    <div className='tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5'>
      <picture>
        <source
          type="image/avif"
          srcSet={`
            https://robohash.org/${id}.avif 200w,
            https://robohash.org/${id}.avif 300w,
            https://robohash.org/${id}.avif 400w,
          `}
          width={'200px'}
          height={'200px'}
        />
        <img
          className={'robot-image'}
          alt='robots'
        />
      </picture>
      <div>
        <h2>{name}</h2>
        <p>{email}</p>
      </div>
    </div>
  );
}

export default Card;
