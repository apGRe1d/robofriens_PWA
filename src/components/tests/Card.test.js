import { shallow, mount, render } from 'enzyme';
import Card from '../Card';
import React from 'react';
import toJson from 'enzyme-to-json';
import { shallowToJson } from "enzyme-to-json";

it('expect to render Card component ', function () {
  // expect(shallow(<Card />).length).toEqual(1);
  // expect(shallowToJson(shallow(<Card />))).toMatchSnapshot();
  expect(shallow(<Card />)).toMatchSnapshot();
  // expect(toJson(shallow(<Card />))).toMatchSnapshot();
  // expect(shallow(<Card />).getElement()).toMatchSnapshot();
});
console.log(shallow(<Card />));
