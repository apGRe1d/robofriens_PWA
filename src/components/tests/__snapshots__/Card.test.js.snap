// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card component  1`] = `
<div
  className="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
>
  <picture>
    <source
      height="200px"
      srcSet="
            https://robohash.org/undefined.avif 200w,
            https://robohash.org/undefined.avif 300w,
            https://robohash.org/undefined.avif 400w,
          "
      type="image/avif"
      width="200px"
    />
    <img
      alt="robots"
      className="robot-image"
    />
  </picture>
  <div>
    <h2 />
    <p />
  </div>
</div>
`;
